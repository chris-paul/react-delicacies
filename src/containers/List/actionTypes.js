/*
 * @Author: 廉恒凯
 * @Date: 2020-04-27 21:15:15
 * @LastEditors: 廉恒凯
 * @LastEditTime: 2020-05-06 13:19:52
 * @Description: file content
 */
const FETCH_COUNTERLIST_LOADING = 'FETCH_COUNTERLIST_LOADING';

const FETCH_COUNTERLIST_FAILURE = 'FETCH_COUNTERLIST_FAILURE';

const FETCH_COUNTERLIST_SUCCESS = 'FETCH_COUNTERLIST_SUCCESS';

const INCREMENT = 'INCREMENT';

const DECREMENT = 'DECREMENT';

export {
    DECREMENT,
    INCREMENT,
    FETCH_COUNTERLIST_LOADING,
    FETCH_COUNTERLIST_FAILURE,
    FETCH_COUNTERLIST_SUCCESS,
};
