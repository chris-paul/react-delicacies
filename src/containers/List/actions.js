/*
 * @Author: 廉恒凯
 * @Date: 2020-04-19 14:55:12
 * @LastEditors: 廉恒凯
 * @LastEditTime: 2020-09-13 16:53:54
 * @Description: file content
 */
import {
    FETCH_COUNTERLIST_LOADING,
    FETCH_COUNTERLIST_FAILURE,
    FETCH_COUNTERLIST_SUCCESS,
    INCREMENT,
    DECREMENT,
} from './actionTypes';
import getInitList from '../../server';

export const increment = caption => ({
    type: INCREMENT,
    caption,
});

export const decrement = caption => ({
    type: DECREMENT,
    caption,
});

export const getCounterList = () => {
    return dispatch => {
        dispatch({
            type: FETCH_COUNTERLIST_LOADING,
        });
        getInitList()
            .then(res => {
                dispatch({
                    type: FETCH_COUNTERLIST_SUCCESS,
                    counterList: res.data,
                });
            })
            .catch(error => {
                dispatch({
                    type: FETCH_COUNTERLIST_FAILURE,
                    errorMessages: error.message || error,
                });
            });
    };
};
