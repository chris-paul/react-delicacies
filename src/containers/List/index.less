.counterListWrap {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 0 auto;
}
