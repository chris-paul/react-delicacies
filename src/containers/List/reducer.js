/*
 * @Author: your name
 * @Date: 2019-12-28 10:18:39
 * @LastEditTime: 2020-05-08 22:37:12
 * @LastEditors: 廉恒凯
 * @Description: In User Settings Edit
 * @FilePath: \react-delicacies\src\reducers\rootReducers.js
 */
/*  项目根目录reducer */
import { fromJS } from 'immutable';
import { createReducer } from 'redux-immutablejs';
import {
    FETCH_COUNTERLIST_LOADING,
    FETCH_COUNTERLIST_FAILURE,
    FETCH_COUNTERLIST_SUCCESS,
    INCREMENT,
    DECREMENT,
} from './actionTypes';

const initialState = fromJS({
    counterList: [],
});

export default createReducer(initialState, {
    [INCREMENT]: (state, { caption }) => {
        return state.update('counterList', listItem => {
            return listItem.map(row => {
                if (row.get('caption') === caption) {
                    return row.set('value', row.get('value') + 1);
                }
                return row;
            });
        });
    },
    [DECREMENT]: (state, { caption }) => {
        return state.update('counterList', listItem => {
            return listItem.map(row => {
                if (row.get('caption') === caption) {
                    return row.set('value', row.get('value') - 1);
                }
                return row;
            });
        });
    },
    [FETCH_COUNTERLIST_LOADING]: state => {
        return state.set('isFetching', true);
    },
    [FETCH_COUNTERLIST_FAILURE]: (state, { errorMessages }) => {
        return state.set('isFetching', false).set('errorMessages', errorMessages);
    },
    [FETCH_COUNTERLIST_SUCCESS]: (state, { counterList }) => {
        console.info(counterList);
        return state.set('isFetching', false).set('counterList', fromJS(counterList));
    },
});
