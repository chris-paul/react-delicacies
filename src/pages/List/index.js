/*
 * @Author: your name
 * @Date: 2019-12-28 10:16:47
 * @LastEditTime : 2020-01-05 19:41:50
 * @LastEditors  : Please set LastEditors
 * @Description: In User Settings Edit
 * @FilePath: \react-delicacies\src\pages\Login\index.js
 */

import { Component } from 'react';
import list from '@api';

class List extends Component {
    async componentDidMount() {
        const { data } = await list.getInitList();
        // eslint-disable-next-line no-console
        console.info(data);
    }

    render() {
        return 'list';
    }
}
export default List;
