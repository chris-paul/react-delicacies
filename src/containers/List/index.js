/*
 * @Author: 廉恒凯
 * @Date: 2019-12-28 10:16:47
 * @LastEditTime: 2021-05-15 12:42:11
 * @LastEditors: 廉恒凯
 * @Description: In User Settings Edit
 * @FilePath: \react-delicacies\src\pages\Login\index.js
 */
import React, { PureComponent } from 'react';
import CounterPanel from '@components/CounterPanel';
import Summary from '@components/Summary';
import FetchError from '@components/FetchError';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import toJS from '@components/Hoc/toJSHOC';
import * as actions from './actions';
import styles from './index.less';

export class List extends PureComponent {
    componentDidMount() {
        const { getCounterList } = this.props;
        getCounterList();
    }

    render() {
        const { isFetching, errorMessages, getCounterList } = this.props;
        if (isFetching) {
            return <p>Loading...</p>;
        }

        if (errorMessages) {
            return (
                <FetchError
                    message={errorMessages}
                    onRetry={() => {
                        getCounterList();
                    }}
                />
            );
        }

        return (
            <div className={styles.counterListWrap}>
                <CounterPanel />
                <Summary />
            </div>
        );
    }
}

List.defaultProps = {
    isFetching: false,
    errorMessages: null,
    getCounterList: () => {},
};

List.propTypes = {
    isFetching: PropTypes.bool,
    errorMessages: PropTypes.string,
    getCounterList: PropTypes.func,
};

export default connect(
    state => {
        return {
            isFetching: state.getIn(['list', 'isFetching']),
            errorMessages: state.getIn(['list', 'errorMessages']),
        };
    },
    dispatch => ({ ...bindActionCreators(actions, dispatch) }),
)(toJS(List));
