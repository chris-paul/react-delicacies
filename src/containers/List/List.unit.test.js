/*
 * @Author: 廉恒凯
 * @Date: 2020-04-18 15:38:31
 * @LastEditors: 廉恒凯
 * @LastEditTime: 2020-04-19 13:32:31
 * @Description: 单元测试 TDD 测试先行 逐步完善代码
 */
import React from 'react';
import { shallow } from 'enzyme';
import List from '.';

describe(' List Component ', () => {
    test('<List /> mounts without crashing', () => {
        const wrapper = shallow(<List />);
        expect(wrapper).toMatchSnapshot();
    });
});
