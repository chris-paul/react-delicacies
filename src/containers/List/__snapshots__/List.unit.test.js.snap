// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` List Component  <List /> mounts without crashing 1`] = `
<div>
  <CounterPanel />
  <CounterPanel />
</div>
`;
